### PR TITLE
Base driver test

### PR DESCRIPTION
**Pull request by yoshiyahoo**

### CODE REVIEW

**Review by Palak-Guleria — DISMISSED** [review]

**Comment by MichaelCrandall** [comment]
You must resolve these conflicts before I can merge your branch

File: .idea/gradle.xml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (MichaelCrandall) [inline .idea/gradle.xml:1]: dont change this file

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -18,8 +18,8 @@
  * project.
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/Robot.java:18]: dont change this file

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -7,34 +7,80 @@
...
+  private final Outtake m_outtake = new Outtake();
+  private final ArcadeDrive m_arcadeDrive = new ArcadeDrive(m_driveTrain);
+  private final TankDrive m_tankDrive = new TankDrive(m_driveTrain);
+  private final ShiftUp m_shiftUp = new ShiftUp();
+  private final ShiftDown m_shiftDown = new ShiftDown();
+  private final ExampleCommand m_autoCommand = new ExampleCommand(m_exampleSubsystem);
+  private final RunCompressor m_runCompressor = new RunCompressor(m_driveTrain);
+  private final Park m_park = new Park(m_driveTrain);
+  private final Fire m_fire = new Fire(m_outtake, 5000); //TODO: After the first argument a second may follow, this will be some sort of number, you can put any number in for now\
+    private final GoToBottom m_goToBottom = new GoToBottom(m_climb); //TODO: The first argument will always be the subsystem
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/RobotContainer.java:49]: remove this extra tab

File: src/main/java/frc/robot/commands/ExampleCommand.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*----------------------------------------------------------------------------*/
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/commands/ExampleCommand.java]: we dont want this file

File: src/main/java/frc/robot/commands/Park.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package frc.robot.commands;
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/commands/Park.java:1]: have this use PIDCommand

File: src/main/java/frc/robot/commands/ShiftUp.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    @Override
+    public void initialize(){
+        m_driveTrain.shiftUp();
+    }
+    @Override
+    public void end(boolean interrupted){}
+
+    @Override
+    public boolean isFinished() {
+        return false;
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/commands/ShiftUp.java]: this should be true

File: src/main/java/frc/robot/commands/TankDrive.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    public TankDrive(DriveTrain tankDrive){
+        m_driveTrain = tankDrive;
+    }
+
+    @Override
+    public void initialize(){}
+
+    @Override
+    public void execute(){
+        double m_leftSpeed = RobotContainer.left_joystick.getY();
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/commands/TankDrive.java]: make sure this also changed

File: src/main/java/frc/robot/subsystems/Autonomous.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package frc.robot.subsystems;
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/Autonomous.java:1]: we do not want this file

File: src/main/java/frc/robot/subsystems/DriveTrain.java [hunk 9]
```diff
@@ -32,11 +51,47 @@ public DriveTrain() {
...
+    public void compressorStart(){
+        m_compressor.start();
+    }
+
+    public void compressorStop(){
+        m_compressor.stop();
+    }
+
+    public double getYaw(){
+        double x;
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/DriveTrain.java:68]: are you sure this is how you get the value from limelimelight, I thought .getDouble() was to choice which pipeline we wanted

File: src/main/java/frc/robot/subsystems/ExampleSubsystem.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
+/*----------------------------------------------------------------------------*/
```
Comment (MichaelCrandall) [inline src/main/java/frc/robot/subsystems/ExampleSubsystem.java]: we do not want this file

**Review by MichaelCrandall — CHANGES_REQUESTED** [review]

**Review by MichaelCrandall — APPROVED** [review]

**Pull request merged**